fractal_server/app/routes/auth/group.py

```diff
@@ -174,18 +177,49 @@ async def update_single_group(
     return updated_group
 
 
-@router_group.delete(
-    "/group/{group_id}/", status_code=status.HTTP_405_METHOD_NOT_ALLOWED
-)
+@router_group.delete("/group/{group_id}/", status_code=204)
 async def delete_single_group(
     group_id: int,
     user: UserOAuth = Depends(current_active_superuser),
     db: AsyncSession = Depends(get_async_db),
-) -> UserGroupRead:
-    raise HTTPException(
-        status_code=status.HTTP_405_METHOD_NOT_ALLOWED,
-        detail=(
-            "Deleting a user group is not allowed, as it may restrict "
-            "previously-granted access.",
-        ),
+) -> Response:
+
+    group = await db.get(UserGroup, group_id)
```

Comment:
Extract `_get_usergroup_or_404` from `_get_single_group_with_user_ids`, and use it in the GET/PATCH/DELETE endpoints.

(and rename `_get_single_group_with_user_ids` into `_get_single_usergroup_with_user_ids`)